lib/animina_web/live/beta_register.ex

```diff
@@ -28,6 +31,84 @@ defmodule AniminaWeb.BetaRegisterLive do
     {:ok, socket}
   end
 
+  @impl true
+
+  def handle_params(%{"step" => "select_white_flags"}, _url, socket) do
```

Comment:
There is a lot to DRY and refactor here. green, red and white are more or less the same code.

```elixir
def handle_params(%{"step" => step}, _url, socket) do
  {color, page_title_str, title_str, info_text_str} = step_info(step)

  {:noreply, assign_flags(socket, color, page_title_str, title_str, info_text_str)}
end

defp step_info("select_white_flags") do
  {
    :white,
    "Select your own flags",
    "Choose Your Own Flags",
    "We use flags to match people. You can select red and green flags later. But first tell us something about yourself and select up to %{number_of_flags} flags that describe yourself. The ones selected first are the most important."
  }
end

defp step_info("select_green_flags") do
  {
    :green,
    "Select your green flags",
    "Choose Your Green Flags",
    "Choose up to %{number_of_flags} flags that you want your partner to have. The ones selected first are the most important."
  }
end

defp step_info("select_red_flags") do
  {
    :red,
    "Select your red flags",
    "Choose Your Red Flags",
    "Choose up to %{number_of_flags} flags that you don't want to have in a partner. The ones selected first are the most important."
  }
end

defp assign_flags(socket, color, page_title_str, title_str, info_text_str) do
  language = socket.assigns.language

  socket
  |> assign(:color, color)
  |> assign(:categories, fetch_categories())
  |> assign(:title, with_locale(language, fn -> gettext(title_str) end))
  |> assign(:info_text, with_locale(language, fn -> gettext(info_text_str, number_of_flags: @max_flags) end))
  |> assign(:page_title, with_locale(language, fn -> gettext(page_title_str) end))
  |> assign(:step, "select_flags")
end
```

Reply:
Please double check this. I just used ChatGPT to refactor it. I have never tested it.

Reply:
Okay , I will take this into account in next PRs